### PR TITLE
Release 0.2.1

### PR DESCRIPTION
**Pull request by adrianabedon**
Changes
- Combined checking package and metadata checksums into one functions shared by apt and yum file checks.
- File checksum mismatches reported as one error even if there are multiple checksums for a single file.
- Fixed file small logic error where an error would not occur if only one of two conditions were not met.

### CODE REVIEW

File: repoaudit/repoaudit/apt.py [hunk 1]
```diff
@@ -48,17 +51,15 @@ def _check_apt_repo_metadata(url: str, dist: str, release_file: Release,
...
 
     dist_url = urljoin(url, "dists", dist)
-    checksum_types = {key: CHECKSUMS[key] for key in CHECKSUMS.keys() if key in release_file}
+    checksums = set(CHECKSUMS.keys()) & set(release_file)
 
     success = True
 
-    files = dict()
-    for key in checksum_types:
+    files = defaultdict(lambda: [])
```
Comment (jasonzio) [inline repoaudit/repoaudit/apt.py:58]: Isn't `list` functionally the same as `lambda: []`?

File: repoaudit/repoaudit/utils.py [hunk 2]
```diff
@@ -213,6 +215,60 @@ def check_signature(repo: str, dist: str, file_url: str,
...
+    success = True
+    error_str = f"{file_type.capitalize()} checksum mismatch for '{file_loc_in_repo}': ["
+
+    for alg, expected in expected_checksums:
+        if multihash.hexdigest(alg) != expected:
+            comma = '' if success else ', '
+            error_str += (
+                f"{comma}{alg.upper()} expected '{expected}' "
+                f"but received '{multihash.hexdigest(alg)}'"
+            )
```
Comment (jasonzio) [inline repoaudit/repoaudit/utils.py:261]: The pythonic way to do this is to build a list of individual error message fragments and then use join, i.e.
```
parts = []
for alg, expected...
    if multihash.hexdigest...
        parts.append(f"{alt.upper()} expected ... but received ...")
if parts:
    error_str = f"... checksum mismatch for '{file...}': [{", ".join(parts)}]"
    errors.add(repo, dist, error_str)
    return False

return True
```

**Review by jasonzio — CHANGES_REQUESTED** [review]

File: repoaudit/repoaudit/apt.py (hunk 1 above)
Comment (adrianabedon) [reply repoaudit/repoaudit/apt.py]: Changed to list

File: repoaudit/repoaudit/utils.py (hunk 2 above)
Comment (adrianabedon) [reply repoaudit/repoaudit/utils.py:259]: Yeah that's cleaner. I changed to code to this

**Review by mbearup — APPROVED** [review]

**Review by jasonzio — APPROVED** [review]
LGTM

**Pull request merged**